### PR TITLE
Type Error: Type error engineConnection.ts

### LINKED ISSUE #785: Uncaught TypeError engineConnection.ts
Was poking around the website and triggered a `Uncaught TypeError`. I did it by first triggering `KCLError` then undoing what caused it.

Example going from this 
```js
const sigmaAllow = 15000 // psi
const width = 11 // inch
const p = 150 // Force on shelf - lbs
const distance = 12 // inches
const FOS = 2
const thickness = sqrt(distance * p * FOS * 6 / ( sigmaAllow * width ))
const filletR = thickness * 2
const shelfMountL = 9
const wallMountL = 8   

;

const bracket = startSketchAt([0 , 0])
  |> line([0, wallMountL], %)
  |> tangentalArc({
    radius: filletR,
    offset: 90
  }, %)
  |> line([-shelfMountL, 0], %)
  |> line([0, -thickness], %)
  |> line([shelfMountL, 0], %)
  |> tangentalArc({
    radius: filletR - thickness,
    offset: -90
  }, %)
  |> line([0, -wallMountL], %)
  |> close(%)
  |> extrude(width, %)

show(bracket)
```

to this

```js
const sigmaAllow = 15000 // psi
const width = 11 // inch
const p = 150 // Force on shelf - lbs
const distance = 12 // inches
const FOS = 2
const thickness = sqrt(distance * p * FOS * 6 / ( sigmaAllow * width ))
const filletR = thickness * 2
const shelfMountL = 9
const wallMountL = 8   



const bracket = startSketchAt([0 , 0])
  |> line([0, wallMountL], %)
  |> tangentalArc({
    radius: filletR,
    offset: 90
  }, %)
  |> line([-shelfMountL, 0], %)
  |> line([0, -thickness], %)
  |> line([shelfMountL, 0], %)
  |> tangentalArc({
    radius: filletR - thickness,
    offset: -90
  }, %)
  |> line([0, -wallMountL], %)
  |> close(%)
  |> extrude(width, %)

show(bracket)
```
image

https://github.com/KittyCAD/modeling-app/assets/82833619/fc20c394-20fa-4c3a-8fad-f3e07518f6fe

I was able to fix this type error in a fork, is this something I can open up a PR for ? Not sure if there are things that I cannot open up PR's for.

### PR DESCRIPTION
**Pull request by notalfredo**
Fixes #785

### CODE REVIEW

**Comment by jessfraz** [comment]
this seems good and harmless to me, thank you! will leave to @Irev-Dev to be sure tho!

File: src/lang/std/engineConnection.ts [hunk 1]
```diff
@@ -832,9 +832,9 @@ export class EngineCommandManager {
     } else {
       this.artifactMap[id] = {
         type: 'failed',
-        range: command.range,
-        commandType: command.commandType,
-        parentId: command.parentId ? command.parentId : undefined,
+        range: command !== undefined ? command.range : undefined,
```
Comment (adamchalmers) [inline src/lang/std/engineConnection.ts:835]: I am _not_ a Typescript expert, but is the same as 
```suggestion
        range: command?.range
```

**Comment by jessfraz** [comment]
after your first contribution it should stop asking us for perms on ci :)

**Comment by notalfredo** [comment]
> after your first contribution it should stop asking us for perms on ci :)

is that good or bad ? I thought the ci where to stop mistakes from happening ?

**Comment by jessfraz** [comment]
Ci is good! but its asking me for approval to run your code everytime you push haha

**Comment by Irev-Dev** [comment]
Artifacts should always have a range and cmd type, if there are cmds that don't it probably means that we are using the `sendModelingCmd` when we should be using `sendSceneCmd`, but maybe it's a bit more complex than that.

This is a good short-term fix though and really the artifact mapping needs some attention anyway as we need to map both on execution and code-gen without execution, so that is probably a good time to polish out the kinks.

**Review by Irev-Dev — APPROVED** [review]

**Comment by notalfredo** [comment]
Okay looks like im still failing types. Im assuming I need to get those fixed up before we can merge this right ? Also I just realized `yarn tsc` is what catches those and I can check that before pushing. If its the case that I need to fix those type errors before the merge, I can look into more tomorrow afternoon if thats fine.

**Comment by Irev-Dev** [comment]
Typescript checks fail, probably because we're relying on those properties being on `CommandInfo` elsewhere.
Might be good to dig a bit deeper as to what's going wrong instead of changing the types.

I think maybe `handleFailedModelingCommand` is being run for all messages, regardless of weather it was a modeling cmd or scene cmd. I don't think the error message gives us that info, but checking it's existing in the artifact map is probably just as good
image

**Comment by notalfredo** [comment]
> Would you mind trying changing the if statement with the original types and see if that works?
> 
> Those types were not optional for a reason, and I worry it's causing more issues.

> handleFailedModelingCommand

just to make sure I understand this correctly you are saying change to change the if statement to ` else if (!message.success && message.request_id && this.artifactMap[message.request_id]) {` and change the types back to what they where normally ( before I made changes to them )?

**Comment by Irev-Dev** [comment]
Awesome!

**Comment by Irev-Dev** [comment]
Is ubuntu build something we should worry about @pierremtb?

**Comment by Irev-Dev** [comment]
Ah nah that's not a problem, it failed because of the key for the updater, and I'm guessing that's because the secret isn't available for outside PRs. I'm going to merge this one.

Thanks again @notalfredo

**Pull request merged**